src/ComputeSharp.D2D1/Shaders/Interop/D2D1ResourceTextureManager.cs

```diff
@@ -0,0 +1,487 @@
+        fixed (uint* pStrides = strides)
+        {
+            D2D1_RESOURCE_TEXTURE_PROPERTIES d2D1ResourceTextureProperties;
+            d2D1ResourceTextureProperties.extents = pExtents;
+            d2D1ResourceTextureProperties.dimensions = (uint)extents.Length;
+            d2D1ResourceTextureProperties.bufferPrecision = (D2D1_BUFFER_PRECISION)bufferPrecision;
+            d2D1ResourceTextureProperties.channelDepth = (D2D1_CHANNEL_DEPTH)channelDepth;
+            d2D1ResourceTextureProperties.filter = (D2D1_FILTER)filter;
+            d2D1ResourceTextureProperties.extendModes = (D2D1_EXTEND_MODE*)pExtendModes;
+
+            int hresult = ((ID2D1ResourceTextureManager*)resourceTextureManager)->Initialize(
+                resourceId: &resourceId,
+                resourceTextureProperties: &d2D1ResourceTextureProperties,
+                data: pData,
+                strides: pStrides,
+                dataSize: (uint)data.Length);
+
+            Marshal.ThrowExceptionForHR(hresult);
+        }
+    }
+
+    /// <summary>
+    /// Updates the data in a given <c>ID2D1ResourceTextureManager</c> instance.
+    /// </summary>
+    /// <param name="resourceTextureManager">A pointer to the <c>ID2D1ResourceTextureManager</c> instance to use.</param>
+    /// <param name="minimumExtents">The "left" extent of the updates if specified (if empty, the entire texture is updated).</param>
+    /// <param name="maximimumExtents">The "right" extent of the updates if specified (if empty, the entire texture is updated).</param>
+    /// <param name="strides">The stride to advance through the input data.</param>
+    /// <param name="data">The data to be placed into the resource texture.</param>
+    public static void Update(
+        void* resourceTextureManager,
+        ReadOnlySpan<uint> minimumExtents,
+        ReadOnlySpan<uint> maximimumExtents,
+        ReadOnlySpan<uint> strides,
+        ReadOnlySpan<byte> data)
+    {
+        fixed (uint* pMinimumExtents = minimumExtents)
+        fixed (uint* pMaximumExtents = maximimumExtents)
+        fixed (uint* pStrides = strides)
+        fixed (byte* pData = data)
+        {
+            int dimensions = strides.Length + 1;
+
+            // Same validation as in the Update instance method above
+            if (pMinimumExtents is not null &&
+                pMaximumExtents is not null &&
+                (minimumExtents.Length != dimensions || maximimumExtents.Length != dimensions))
+            {
+                Marshal.ThrowExceptionForHR(E.E_INVALIDARG);
+            }
+
+            int hresult = ((ID2D1ResourceTextureManager*)resourceTextureManager)->Update(
+                minimumExtents: pMinimumExtents,
+                maximimumExtents: pMaximumExtents,
+                strides: pStrides,
+                dimensions: (uint)dimensions,
+                data: pData,
+                dataCount: (uint)data.Length);
+
+            Marshal.ThrowExceptionForHR(hresult);
```

Comment:
does this method also need a `GC.KeepAlive(this)` ?

Reply:
See above, this is a static method.